### ISSUE TITLE
bug: 🐛 Mini app does not work with Node 16

### ISSUE THREAD

**Issue by frankstallone** [Cuberto/scroll-sequence-demo#1]
First, thanks for sharing. Great YouTube video and great work! Y'all may want to add `engines` to `package.json` [so others know which version of Node to use](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#engines). I was on 16 but told `nvm` to use `14` and was able to get this working. The error messages on 16 were long when running `npm i` but here are some of them

```bash
...
npm ERR!   c++ '-DNODE_GYP_MODULE_NAME=libsass' '-DUSING_UV_SHARED=1' '-DUSING_V8_SHARED=1' '-DV8_DEPRECATION_WARNINGS=1' '-DV8_DEPRECATION_WARNINGS' '-DV8_IMMINENT_DEPRECATION_WARNINGS' '-D_GLIBCXX_USE_CXX11_ABI=1' '-D_DARWIN_USE_64_BIT_INODE=1' '-D_LARGEFILE_SOURCE' '-D_FILE_OFFSET_BITS=64' '-DOPENSSL_NO_PINSHARED' '-DOPENSSL_THREADS' '-DLIBSASS_VERSION="3.5.5"' -I/Users/chromebook/.node-gyp/16.13.0/include/node -I/Users/chromebook/.node-gyp/16.13.0/src -I/Users/chromebook/.node-gyp/16.13.0/deps/openssl/config -I/Users/chromebook/.node-gyp/16.13.0/deps/openssl/openssl/include -I/Users/chromebook/.node-gyp/16.13.0/deps/uv/include -I/Users/chromebook/.node-gyp/16.13.0/deps/zlib -I/Users/chromebook/.node-gyp/16.13.0/deps/v8/include -I../src/libsass/include  -O3 -gdwarf-2 -mmacosx-version-min=10.7 -arch x86_64 -Wall -Wendif-labels -W -Wno-unused-parameter -std=c++11 -stdlib=libc++ -fno-strict-aliasing -MMD -MF ./Release/.deps/Release/obj.target/libsass/src/libsass/src/ast.o.d.raw   -c -o Release/obj.target/libsass/src/libsass/src/ast.o ../src/libsass/src/ast.cpp
...
npm ERR! In file included from ../src/libsass/src/ast.cpp:2:
npm ERR! ../src/libsass/src/ast.hpp:1614:25: warning: loop variable 'numerator' creates a copy from type 'const std::string' [-Wrange-loop-construct]
npm ERR!         for (const auto numerator : numerators)
...

```

**Comment by orgildinio** [comment]
Please update dependencies. i`m getting same error as @frankstallone

**Comment by Drafteed** [comment]
Should be fixed in b0c7945

**Comment by lucswart** [comment]
Still bugs:
```
gulp-sass no longer has a default Sass compiler; please set one yourself.
Both the "sass" and "node-sass" packages are permitted.
For example, in your gulpfile:

  const sass = require('gulp-sass')(require('sass'));
```

**Comment by flowgrow** [comment]
I fixed it by adding the proposed line, and then removing the $. in front of the sass function in line 40:

```
  /*!
  * Gulp SMPL Layout Builder
  *
  * @version 7.3.1 (lite)
  * @author Artem Dordzhiev (Draft) | Cuberto
  * @type Module gulp
  */

  /* Get plugins */
  const gulp = require('gulp');
  const sass = require('gulp-sass')(require('sass'));
  const browserSync = require('browser-sync');
  const $ = require('gulp-load-plugins')({pattern: ['gulp-*', 'gulp.*', 'del']});
  const webpack = require('webpack-stream');

  /* Primary tasks */
  gulp.task('default', (done) => {
      gulp.series('serve')(done)
  });

  gulp.task('serve', (done) => {
      gulp.series('clean', gulp.parallel('pug', 'sass', 'js'), 'browsersync', 'watch')(done)
  });

  /* Pug task */
  gulp.task('pug', () => {
      return gulp.src(['./src/pug/**/*.pug', '!./src/pug/_includes/**/*'])
          .pipe($.plumber())
          .pipe($.pug({
              pretty: true,
              basedir: "./src/pug/"
          }))
          .pipe(gulp.dest('./tmp/')).on('end', () => {
              browserSync.reload();
          });
  });

  /* Sass task */
  gulp.task('sass', () => {
      return gulp.src('./src/scss/main.scss')
          .pipe(sass({
              "includePaths": "node_modules"
          }))
          .pipe($.autoprefixer())
          .pipe(gulp.dest('./tmp/assets/css/'))
          .pipe(browserSync.stream({match: '**/*.css'}));
  });

  /* JS task */
  gulp.task('js', () => {
      return gulp.src(['./src/js/**/*'])
          .pipe(webpack(require('./webpack.config.js')))
          .pipe(gulp.dest('./tmp/assets/js'));
  });

  /* Browsersync Server */
  gulp.task('browsersync', (done) => {
      browserSync.init({
          server: ["./tmp", "./src/static"],
          notify: false,
          ui: false,
          online: false,
          ghostMode: {
              clicks: false,
              forms: false,
              scroll: false
          }
      });
      done();
  });

  /* Watcher */
  gulp.task('watch', () => {
      global.isWatching = true;

      gulp.watch("./src/scss/**/*.scss", gulp.series('sass'));
      gulp.watch("./src/pug/**/*.pug", gulp.series('pug'));
      gulp.watch("./src/js/**/*.*", gulp.series('js'));
      gulp.watch("./config.json", gulp.parallel('pug', 'js'));
  });

  /* FS tasks */
  gulp.task('clean', () => {
      return $.del(['./tmp/**/*'], {dot: true});
  });

```

**Comment by kode369** [comment]
the image animation is not working

**Comment by Drafteed** [comment]
Thanks everyone for the report! Problem with sass peer-dependency was fixed in 3d88516
